kiwi_keg/generator.py

```diff
@@ -92,6 +92,19 @@ def create_kiwi_description(
         kiwi = KiwiDescription(outfile)
         if markup == 'xml':
             kiwi.create_XML_description(outfile)
+            # KIWI does not preserve comment blocks after validation.
+            # For OBS the comments are no comments but effective
+            # project config data. Questionable design but we can't
+            # influence this and add back the header comments.
+            #
+            # This is only done for the XML markup since I think
+            # OBS is expecting the comments in XML comment syntax
+            # and does not support anything else. If this turns
+            # out to be an issue in the future and people start
+            # to use YAML markup for KIWI images in OBS we need
+            # add a translation of the XML comments to YAML
+            # after create_YAML_description
+            self._preserve_XML_comments(kiwi_document, outfile)
```

Comment:
I wonder if `KegGenerator` should preserve the comments or it is a `KiwiDescription responsability`
Maybe after this [line](https://github.com/SUSE-Enceladus/keg/blob/main/kiwi_keg/kiwi_description.py#L64)
so it could be 
```python
def create_XML_description(self, output_file: str):
        self._create_description(output_file, 'xml')
        self._preserve_XML_comments(params)
```

Reply:
you are right it's a KiwiDescription responsibility. I moved the code. Thanks much